internal/pkg/addon/package.go

```diff
@@ -0,0 +1,302 @@
+			ForceZip:           true,
+		},
+	},
+	"AWS::AppSync::GraphQLSchema": {
+		{
+			Property: []string{"DefinitionS3Location"},
+		},
+	},
+	"AWS::AppSync::Resolver": {
+		{
+			Property: []string{"RequestMappingTemplateS3Location"},
+		},
+		{
+			Property: []string{"ResponseMappingTemplateS3Location"},
+		},
+	},
+	"AWS::AppSync::FunctionConfiguration": {
+		{
+			Property: []string{"RequestMappingTemplateS3Location"},
+		},
+		{
+			Property: []string{"ResponseMappingTemplateS3Location"},
+		},
+	},
+	"AWS::ElasticBeanstalk::ApplicationVersion": {
+		{
+			Property:           []string{"SourceBundle"},
+			BucketNameProperty: "S3Bucket",
+			ObjectKeyProperty:  "S3Key",
+		},
+	},
+	"AWS::Glue::Job": {
+		{
+			Property: []string{"Command", "ScriptLocation"},
+		},
+	},
+	"AWS::StepFunctions::StateMachine": {
+		{
+			Property:           []string{"DefinitionS3Location"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+		},
+	},
+	"AWS::CodeCommit::Repository": {
+		{
+			Property:           []string{"Code", "S3"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+			ForceZip:           true,
+		},
+	},
+}
+
+func (a *Addons) packageLocalArtifacts(tmpl *cfnTemplate) (*cfnTemplate, error) {
+	resources := mappingNode(&tmpl.Resources)
+
+	for name := range resources {
+		res := mappingNode(resources[name])
+		typeNode, ok := res["Type"]
+		if !ok || typeNode.Kind != yaml.ScalarNode {
+			continue
+		}
+
+		propsNode, ok := res["Properties"]
+		if !ok || propsNode.Kind != yaml.MappingNode {
+			continue
+		}
+
+		transforms, ok := transformInfoFor[typeNode.Value]
+		if !ok {
+			continue
+		}
+
+		for _, tr := range transforms {
+			if err := a.transformProperty(propsNode, tr); err != nil {
+				return nil, fmt.Errorf("transform property %s property for %s: %w", name, tr.Property, err)
+			}
+		}
+	}
+
+	return tmpl, nil
+}
+
+func (a *Addons) transformProperty(properties *yaml.Node, tr transformInfo) error {
+	mapNode := mappingNode(properties)
+	var node *yaml.Node
+	for i, key := range tr.Property {
+		var ok bool
+		node, ok = mapNode[key]
+		if !ok || (i+1 != len(tr.Property) && node.Kind != yaml.MappingNode) {
+			return nil // no error if the property doesn't exist
+		}
+		mapNode = mappingNode(node)
+	}
+
+	if node == nil || node.Kind != yaml.ScalarNode {
+		// only transform if the node is a scalar node
+		return nil
+	}
+
+	if strings.HasPrefix(node.Value, "s3://") || strings.HasPrefix(node.Value, "http://") || strings.HasPrefix(node.Value, "https://") {
+		return nil
+	}
+
+	assetPath := node.Value
+	if !path.IsAbs(node.Value) {
+		assetPath = path.Join(a.wsPath, node.Value)
+	}
+
+	url, err := a.uploadAddonAsset(assetPath, tr.ForceZip)
+	if err != nil {
+		return fmt.Errorf("upload asset: %w", err)
+	}
+
+	bucket, key, err := s3.ParseURL(url)
+	if err != nil {
+		return fmt.Errorf("parse s3 url: %w", err)
+	}
+
+	fmt.Printf("Uploaded %s to s3 at: %s\n", assetPath, s3.Location(bucket, key))
```

Comment:
both the aws cli and sam cli print out where they are uploading files...it's definitely not a need, but i thought it might be nice. i'm fine to remove it if you think it's better to not surface it.